src/main/java/org/assertj/core/internal/Paths.java

```diff
@@ -496,4 +499,22 @@ private static void assertExpectedEndPathIsNotNull(final Path end) {
     requireNonNull(end, "the expected end path should not be null");
   }
 
+  public void assertIsEmptyPath(AssertionInfo info, Path actual) {
+    assertIsRegularFile(info, actual);
+    try {
+      if (size(actual) == 0) return;
+    } catch (IOException e) {
+
+    }
+    throw failures.failure(info, shouldBeEmpty(actual));
+  }
+  public void assertIsNotEmptyPath(AssertionInfo info, Path actual){
+    assertIsRegularFile(info, actual);
+    try {
+      if (size(actual) > 0) return;
+    } catch (IOException e) {
+
```

Comment:
Instead of leaving the `catch` empty, you can throw a `PathsException`:
```suggestion
      throw new PathsException(FAILED_TO_RESOLVE_ACTUAL_REAL_PATH, e);
```